Sources/ITwinMobile/ITMGeoLocationManager.swift

```diff
@@ -227,71 +237,74 @@ public class ITMGeolocationManager: NSObject, CLLocationManagerDelegate, WKScrip
+                        try await getCurrentPosition(jsonObject)
+                    default:
+                        ITMApplication.logger.log(.error, "Unknown Bentley_ITMGeolocation messageName: \(messageName)")
+                    }
+                } catch {
+                    ITMApplication.logger.log(.error, "\(messageName) error: \(error.localizedDescription)")
+                }
             }
         } else {
             ITMApplication.logger.log(.error, "Bentley_ITMGeolocation messageName is not a string.")
         }
     }
 
     private func updateHeadingOrientation() {
-        var orientation: CLDeviceOrientation
-        switch UIDevice.current.orientation {
-        case .landscapeLeft:
-            orientation = .landscapeLeft
-            break
-        case .landscapeRight:
-            orientation = .landscapeRight
-            break
-        case .portrait:
-            orientation = .portrait
-            break
-        case .portraitUpsideDown:
-            orientation = .portraitUpsideDown
-            break
-        default:
-            return
+        let deviceOrientation = UIDevice.current.orientation
+        let orientation: CLDeviceOrientation = switch deviceOrientation {
+        case .landscapeLeft: .landscapeLeft
+        case .landscapeRight: .landscapeRight
+        case .portrait: .portrait
+        case .portraitUpsideDown: .portraitUpsideDown
+        default: .unknown // face up and face down are ignored by CLLocationManager
         }
-        if locationManager.headingOrientation != orientation {
-            locationManager.headingOrientation = orientation
-            if !watchIds.isEmpty {
-                // I'm not sure if this is necessary or not, but it can't hurt.
-                // Note that to force an immediate heading update, you have to
-                // call stop then start.
-                locationManager.stopUpdatingHeading()
-                locationManager.startUpdatingHeading()
-            }
+        guard orientation != .unknown, locationManager.headingOrientation != orientation else { return }
+        locationManager.headingOrientation = orientation
+        if !watchIds.isEmpty {
+            // I'm not sure if this is necessary or not, but it can't hurt.
+            // Note that to force an immediate heading update, you have to
+            // call stop then start.
+            locationManager.stopUpdatingHeading()
+            locationManager.startUpdatingHeading()
         }
     }
+    
+    /// Determine if the given permission indicates the user is authorized to check location.
+    ///
+    /// Override this function in a subclass in order to add custom behavior.
+    /// - Parameter permission: The `CLAuthorizationStatus` to check.
+    /// - Returns: By default, returns `true` if `permission` is `authorizedAlways` or `authorizedWhenInUse`,
+    /// and `false` otherwise.
+    public class func isAuthorized(_ permission: CLAuthorizationStatus) -> Bool {
```

Comment:
This also gets called by AsyncLocationManager's function, and there it has to use ITMGeolocationManager, not Self. Because of this, I think this change would just cause problems.

Reply:
OK, so we should remove the comment about overriding it in a subclass then.

Reply:
Done, and also changed to static instead of class.